client/client.go

```diff
@@ -140,6 +140,7 @@ func WithRetry(retry uint64) RegionsOption {
 
 type tsoRequest struct {
 	start      time.Time
+	client     *client
```

Comment:
Maybe we can just pass the `client.ctx` here rather than the whole `client`? It's more specific and clear.